parquet/src/main/java/org/apache/iceberg/parquet/ReadConf.java

```diff
@@ -181,8 +184,8 @@ boolean[] shouldSkip() {
     return shouldSkip;
   }
 
-  private Map<Long, Long> generateOffsetToStartPos(Schema schema) {
-    if (schema.findField(MetadataColumns.ROW_POSITION.fieldId()) == null) {
+  private Map<Long, Long> generateOffsetToStartPos() {
+    if (hasPositionDelete) {
```

Comment:
I'd recommend to do it in the caller like this 
```
    Map<Long, Long> offsetToStartPos = hasPositionDelete? generateOffsetToStartPos(): null;
```

Reply:
BTW, we don't actually need this if this PR#10107 is in, which removes related complexity. The parquet side is ready  for a while. We may get #10107 in.